cpp/src/arrow/dataset/file_base.h

```diff
@@ -387,6 +387,9 @@ struct ARROW_DS_EXPORT FileSystemDatasetWriteOptions {
   /// Controls what happens if an output directory already exists.
   ExistingDataBehavior existing_data_behavior = ExistingDataBehavior::kError;
 
+  /// Flag to restrict or allow creating directory for writing the dataset.
```

Comment:
```suggestion
  /// Whether to attempt creating the dataset directory.
  /// This can be set to false to work around limited permissions on some filesystems.
```

Reply:
Made the change, thanks!